Trigger a build

milljm opened the pull request:
this is needed once again.


Testing meowbuild.

dschwen committed:
Approximate gap derivatives (#13488)

Commit:
fix and closes #15793

recuero committed:
Initial commit for anisotropic plasticity Refs. #16016

Pass tensors by reference

Use DenseVector and DenseMatrix

Move computeReferenceResidual to the material. More general

Update inelastic strain tensor upon convergence of plastic multiplier (orthotropic creep model).

Move to an AD hierarchy

New AD hierarchy for anisotropic plasticity

Toy problem for anisotropic plasticity 'running'

Anisotropic creep model converges, but does not creep.

Simple 2D model working (anis_mech_hill_tensor_creep_small.i). Next plasticity and verification.

Create Hill creep model 2D tests

Add anisotropic plasticity classes

Add gold files, remove banned keyword

Fix non-unity build

Update anisotropy PR to new generic RadialReturn class

Anisotropic plasticity based on generalized radial return mapping

Problem converges in few iterations (doesn't modify stress or strain)

New features:

 - Make material time step work through effective_inelastic_strain
 - Make call to computeStressFinalize to modify Cauchy stress due to plastic yield
 - Disable debugging Moose::out statements

Add plastic tests that ensure:

 - Anisotropic plastic test behaves like isotropic elastic if no plastic excusion takes place
 - Evolution of plastic strain and convergence looks okay by simple inspection (verification ahead)

Avoid compiler warnings for unused variables.

Demote sanity check tests to first order to prevent having insufficient AD vector size

recuero committed:
Add markdown documentation to anisotropic creep and plasticity

Commit:
Trigger a build

Do not merge.

Refs #000

moosebuild commented:
Job [(meowbuild) Documentation](https://civet.inl.gov/job/658549/) on f868e8c wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/16683/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/658535/) on f868e8c wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/16683/site)

This comment will be updated on new commits.

moosebuild commented:
Job [(meowbuild) Documentation](https://civet.inl.gov/job/658549/) on f868e8c wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/16683/site)

This comment will be updated on new commits.

moosebuild commented:
Job [(meowbuild) Documentation](https://civet.inl.gov/job/658549/) on f868e8c wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/16683/site)

This comment will be updated on new commits.

Commit:
Trigger a build.

Testing combined mac/end user test target.

Refs #000

moosebuild commented:
All [jobs](https://civet.inl.gov/event/62279/) on 5dab5a2 : canceled by @milljm

Need to update recipes...

Commit:
Trigger a build

Refs #000

Commit:
trigger a build...

With the recipe in place this time... trigger a build.

recuero committed:
Fix accumulation of isotropic hardening variable

recuero committed:
Add verification-based regression tests for isotropic elasticity

recuero committed:
Fix tests in parallel

Some code cleanup, doxygen comments

recuero committed:
Address reviewer's comment

Address reviewer's comments

recuero committed:
Move definition of eigendecomposition to MooseTypes, simplify input files

Clean up existing hierarchy

GiudGiud committed:
Allow use of FV variables with neighborCoupledValue/Gradient

recuero committed:
Initial implementation/verification of elastoplastic anisotropy

recuero committed:
Change class names to 'Hill'

recuero committed:
Fix creep plastic multiplier bug and update tests

recuero committed:
Add Hill creep tests from analytical verification exercise

Avoid Eigen's ambiguous operator error.

- Eigen matrices will be non-AD. For now, they can be considered as user input.
- Some clean up. Apply hit format to some inputs.

Commit:
Trigger a build

lindsayad committed:
Execute mortar constraints before residual/matrix assembly

lindsayad committed:
Two-pass implementation of weighted gap LM mortar constraint

Closes #16961

lindsayad committed:
Demonstrate how to use two-pass objects

lindsayad committed:
Prevent repeat failed projections of primary nodes

lindsayad committed:
If reached max iterations, consider projection failed

lindsayad committed:
Don't call dof_number if not n_dofs

lindsayad committed:
Mixed basis test

lindsayad committed:
Documentation

lindsayad committed:
Remove need to call NonlinearSystemBase::mortarConstraints twice

It's more transparent just to have a single call to `mortarConstraints`
that executes both undisplaced and displaced mortar constraints. The old
way was motivated by having computation of mortar constraints occur in
`constraintResiduals/constraintJacobians` but we now compute mortar
constraints ahead of those methods in order to ensure that mortar
contributions are assembled into the global residual/Jacobian before we
evaluate other constraints

recuero committed:
Make anisotropic creep truly implicit - Before this commit, elastic strain was not fully correct - Update tests, successful comparison with isotropic power law - Address reviewer's comment on the use of "zero()" method

Anisotropic creep is sensitive to time step and stress increments. To be investigated.

lindsayad committed:
Make lm diagonal entry positive instead of negative

lindsayad committed:
Add min_gap parameter to GapConductanceConstraint

lindsayad committed:
Add tests with production tensor mechanics objects plus mechanical contact (#13488)

Complete test matrix:

2x2x2x2 [low-k, high-k] x [first order, second order] x [small,finite]
          x [cartesian, RZ]

lindsayad committed:
Protect against deriv insertion for node without active displacements

tophmatthews committed:
Doco fixes for Modules/HeatConduction syntax
Closes #17259

tophmatthews committed:
Move tests

tophmatthews committed:
Remove density from ThermalContact tests

tophmatthews committed:
Create new ThermalContact syntax

lindsayad committed:
Solve nodal projection tighter

It doesn't make sense to me to solve the (non)linear projection system
with the same tolerance as the comparison tolerance we're going to use
when deciding, for example, whether we should have already established a
map between a secondary and primary node. In general we always do our
solve tolerances tighter than our comparison tolerances

lindsayad committed:
Restore single way of computing normal field for mortar

We revert to the initial John Peterson implementation of normal fields
where we use the secondary nodal normals both for projecting secondary
and primary nodes. This seems to avoid issues of different projection
behavior between the sides (duh)

lindsayad committed:
Regold failing contact tests after changing primary node projection

We changed the projection method for primary nodes in a previous commit.
More specifically we now use the same normal field for projecting
primary nodes as we do for projecting secondary nodes, with the normal
field corresponding to interpolation of secondary nodal normals. This
change was made with the target of creating a consistent mortar mesh,
e.g as an example we should never run into the case where a primary node
projects onto a secondary node, but the secondary node doesn't project
onto the primary node (or visa versa). We ran into this issue quite a
bit in a fairly simple input that Antonio sent me, but the change to a
single normal projection field fixed the issue (or made it a lot better
at a minimum).

With the projection method changed, we have to regold some contact tests

recuero committed:
Partial work for creep strain rate integration time step

permcody committed:
Make "moosetools" optional - sortof. HIT is still required

refs #17108

permcody committed:
Update framework/moose.mk with better missing HIT message.

Co-authored-by: Andrew E Slaughter <slaughter98@gmail.com>

lindsayad committed:
Add boundaries_to_not_force param to FVFluxKernel

jessecarterMOOSE committed:
moving phase field docs over from old wiki, ref#14988

jessecarterMOOSE committed:
use images from large_media submodule

recuero committed:
Add test for creep rate integration error with soft terminator

lingzou-anl committed:
Address ArrayVariable restart issue

Closes #17121

lindsayad committed:
Prevent strong chance of failed physical point inversion

Right now when someone calls `FEProblemBase::reinitElemPhys` with a
vector of physical points, we will attempt to invert those physical
points into the reference element space for both undisplaced and
displaced meshes if the latter exists. It is not difficult to imagine
that for a given set of physical points you may easily fall off the
displaced mesh for large enough displacements/deformations. The same
issues exists for `FEProblemBase::reinitNeighborPhys`. Callers of these
phys methods should call through the appropriate `SubProblem` that is
relevant to their context. Moreover, we should always make sure that the
caller is calling with an element corresponding to the `SubProblem`
they've selected.

Closes #17355

Commit:
Merge pull request #17317 from loganharbour/more_remove_legacy

Remove preset BC objects and legacy dirichlet BC param

aeslaughter committed:
Revert this to remove HIT...again

aeslaughter committed:
Update Makefile to support framework or moosetools HIT

(refs #17006)

aeslaughter committed:
Update moosetools submodule
(refs #17006)

aeslaughter committed:
Use udate_and_rebuild_hit in Makefile error

(refs #17006)

dschwen committed:
Merge pull request #17285 from recuero/templatedStressUpdateSignature

Unify AD and non-AD StressUpdateBase signatures

lindsayad committed:
Protect seeding of derivative vector in getElemValue

This will help maintain the integrity of the AD-computed Jacobian if
users use aux variables in their residual computations.

Closes #17375

loganharbour committed:
Remove _u_old from ScalarKernel

refs #17379

loganharbour committed:
Merge pull request #17265 from WilkAndy/demunge_doco_17251

PorousFlow documentation demunged

loganharbour committed:
Merge pull request #16725 from GiudGiud/PR_smol

Add explanation of csv file format in PiecewiseBilinear function

Commit:
Merge pull request #16835 from aprilnovak/file-combiner

Allow CombinerGenerator to take a file of positions.

dschwen committed:
Merge pull request #17352 from dschwen/functionexpression_17325

Set custom function expression type

aeslaughter committed:
Apply suggestions from code review

Co-authored-by: Logan Harbour <loganhharbour@gmail.com>

lindsayad committed:
Merge pull request #17376 from lindsayad/seeding-17375

Protect seeding of derivative vector in getElemValue

smharper committed:
Interpolate the mixing length to element faces

smharper committed:
clang-format

smharper committed:
Set DoF values directly in BuildArrayVariableAux

aeslaughter committed:
Make gitmodules use relative URL

(refs #17006)

recuero committed:
Update updateState signature to ADStressUpdateBase

recuero committed:
Make most creep tests not to consider integration error.

This is done by explicitly prescribing a large value in the input file.
Otherwise, the default value will ensure the output of the material
time step limit captures such integration error.

recuero committed:
Avoid divide-by-zero when computing time step limit

permcody committed:
Merge pull request #17108 from aeslaughter/moosettools-submodule-17006

Add moosetools submodule

smharper committed:
Address PR #16403 comments

lindsayad committed:
Add scaling vector if we are doing automatic scaling

If we have automatic scaling with AD, we need to make sure that we
create the scaling_factors vector or else we end up trying to read
a `scaling_factors` vector that does not yet exist when reading our
restartable data.

This maybe closes #17078

roystgnr committed:
Fix -Wunused-result warning

With GCC 10.2 I get a warning if we don't test the return value of a
chdir() attempt.

Refs #10733

smharper committed:
Add exception tests for BuildArrayVariableAux

andrsd committed:
Fix suppressing of controllable parameters

Unmark parameter as controllable when suppressing it.

Closes #17403

smharper committed:
Remove BuildArrayVariableAux redundant assignment

smharper committed:
Clean up BuildArrayVariableAux

roystgnr committed:
Fix misleading comment

These names can be any mesh generator in the input file, they're not
mesh filenames.

roystgnr committed:
Extend StitchedMeshGenerator code and clone test

The modified test will instantiate "subgenerators" itself instead of
getting them from earlier in the input file.

recuero committed:
Update creep time step documentation

aeslaughter committed:
Fix wildcard logic for HIT_DIR

(refs #17108)

aeslaughter committed:
Correct pyhit logic and use quotes for wildcard checks in moose.mk

(refs #17108)

dschwen committed:
Merge pull request #17261 from tophmatthews/heatconduction_doco_17259

Add Modules/HeatConduction/ThermalContact syntax

aeslaughter committed:
Merge pull request #17415 from aeslaughter/moosetools-hit-make-with-x-17108

Moosetools hit Makefile fix

roystgnr committed:
Merge pull request #17400 from roystgnr/warning_fix

Fix -Wunused-result warning

aeslaughter committed:
Add known location for HIT when loading pyhit

(refs #17108)

lindsayad committed:
Test automatic scaling in sub-applications

Refs #17078

andrsd committed:
Merge pull request #17419 from aeslaughter/moosetools-hit-make-with-x-17108

Add known location for HIT when loading pyhit

dschwen committed:
Merge pull request #17392 from recuero/anisotropy

Anisotropic creep and plasticity

andrsd committed:
Merge pull request #17404 from andrsd/suppress-controllable-param

Fix suppressing of controllable parameters

WilkAndy committed:
GeochemistryUnitConverter utility added, along with unit tests
Refs #17155

WilkAndy committed:
Add notion of units and a new constraint_meaning interface to GeochemicalSystem

The old constraint system had unit choices built in, so with this new unit functionality, the constraint_meaning names did not make sense.  Hence, a new user interface for the constraint_meaning has been created.  It is both simpler and more flexible because units and meanings are now different things, rather than all lumped together.  The new user-interface constraint_meanings are mapped into the old constraint_meanings, during the constructor of GeochemicalSystem, at the same time that the units are all converted to mole-based units

Refs #17155

WilkAndy committed:
tests and documentation of the new geochemistry units stuff

At the same time, various old tests and documentation has been modified to use the new units system.

At the same time, the new HIT notation ([] instead of [../], etc) has been used in all geochemistry tests

Fixes #17155

rwcarlsen committed:
add --copy-tests flag for copying installed tests to curr dir.

This is necessary for people with only level-1 access to various moose
apps/codes - the tests will be installed into a location where they
don't have write access - so the test harness can't run them there.
Running this flag copies the installed tests out to an appname_tests dir
that users can cd into and then run them.

ref #16411

aeslaughter committed:
Correct logic that mandates that pyhit exists in git repo during import

(refs #17108)

hugary1995 committed:
Let DerivativeParsedMaterial properly initQpStatefulProperties

close #17430

hugary1995 committed:
same thing for ParsedMaterialHelper

lindsayad committed:
Clear all dof indices at the beginning of a solve

This prevents bad dof indices from lingering around between
invocations of `FEProblemBase::solve`. Here "bad" dof indices
could mean dof indices that no longer exist due to mesh
adaptivity

roystgnr committed:
Make StitchedMeshGenerator::generate more flexible

If we don't assume that the index names and the mesh ptr vectors are the
same size, then we can have subclasses that add additional mesh ptr
entries from other sources.  This is useful for a test subclass I'm
adding.

aeslaughter committed:
Copy hit.so in install_testharness

(refs #17108)

lindsayad committed:
Add static const getter of MaterialPropertyStorage prop_ids

And then use it in `ComputeFVFluxThread` to supply the name
of the material property that is defined on multiple blocks

lindsayad committed:
Only reinit variables once on boundary faces

roystgnr committed:
Add InputParameters::setParameters() method

Template parameter packs are neat.

More importantly, template parameter packs let us pass arbitrary
parameters to set in the internals of another helper function.

roystgnr committed:
Clarify some MeshGenerator Doxygen comments

roystgnr committed:
Add MeshGenerator::addMeshSubgenerator()

Hopefully this is something like what the ANL folks had in mind for
https://github.com/idaholab/moose/issues/17184

roystgnr committed:
Use addMeshSubgenerator in test code

This turns several lines of slightly tricky code into 4 lines of
arguably self-documenting code, so I'm fairly happy with it.

dschwen committed:
Merge pull request #17431 from hugary1995/derivative_parsed_material_init

Let DerivativeParsedMaterial properly initQpStatefulProperties

YaqiWang committed:
minor optimization in assembly #17448

YaqiWang committed:
add lower-d subdomain for all sides in MooseMesh #17448

aprilnovak committed:
Allow multiple element sides on same sideset for elemental boundary auxkernel. Refs #17424

aprilnovak committed:
Update test/tests/auxkernels/element_aux_boundary/tests

Update wording for requirement

Co-authored-by: Logan Harbour <loganhharbour@gmail.com>

lindsayad committed:
Apply suggestions from code review

Co-authored-by: Logan Harbour <loganhharbour@gmail.com>

lindsayad committed:
Add FVInterfaceKernel

lindsayad committed:
Call SystemBase::subdomainSetup from NonlinearSystemBase::subdomainSetup

rwcarlsen committed:
Merge pull request #17429 from aeslaughter/install-pyhit-17108

Correct logic that mandates that pyhit exists in git repo during import

lindsayad committed:
Make FV methods compilable in non-global-AD configurations

lindsayad committed:
Add FVOneVarDiffusionInterface and test

Demonstrates first order convergence.

Refs #10787

lingzou committed:
Update framework/src/systems/SystemBase.C

Co-authored-by: Logan Harbour <loganhharbour@gmail.com>

lingzou committed:
Update framework/src/systems/SystemBase.C

Co-authored-by: Logan Harbour <loganhharbour@gmail.com>

lindsayad committed:
Merge pull request #17355 from lindsayad/prevent-phys-points-in-elem-disaster

Prevent strong chance of failed physical point inversion

lindsayad committed:
Update libmesh

Summary of changes:

  - Add Tet10 Nedelec
  - Add global data structures for mesh subdomain and boundary ids
  - Deprecate Side, SideEdge proxy classes; add fill-based edge_ptr methods
  - Read/write numeric vector projection/type info
  - Metaphysicl 1.2.2 update
  - Add Elem::has_invertible_map() and overrides for some element types
  - Update TIMPI
  - MeshBase & member in BoundaryInfo changed to MeshBase *
  - MeshFunction: add non-const accessor for underlying PointLocator
  - Elem edge fixes
  - Add a missing pointer to the DofMap's sparsity pattern.
  - Include unpartitioned elements in MeshBase::subdomain_ids
  - Fix matrix_matrix_mult()
  - EquationSystems: store unique_ptrs to Systems we manage
  - Fix the other PETSc version guard
  - modify two unit tests
  - Fix for --enable-petsc-required
  - Add PointLocator implementation based on Nanoflann
  - PointLocatorTree: use std::shared_ptr to manage tree
  - Make MeshRefinement tests const
  - Fix PETSc version guard
  - Mac warning fixes

Refs #000

loganharbour committed:
Make Scalar variables great again with const

refs #17451

loganharbour committed:
Do not get number of components if var is not coupled

refs #17451

smharper committed:
Merge pull request #16403 from smharper/build_array

BuildArrayVariableAux for combining multiple scalar fields into an array field

roystgnr committed:
Document new subgenerator API

lindsayad committed:
Documentation

lindsayad committed:
Add FV variable getter

lindsayad committed:
Apply suggestions from code review

Co-authored-by: Guillaume Giudicelli <guillaume.giudicelli@gmail.com>

roystgnr committed:
Remove legacy params syntax

Thanks to @loganharbour for catching this

roystgnr committed:
Update test/src/meshgenerators/StitchedSubgenerators.C

Make subgenerator name base unique

Co-authored-by: Logan Harbour <loganhharbour@gmail.com>

jessecarterMOOSE committed:
Apply suggestions from code review

Co-authored-by: Logan Harbour <loganhharbour@gmail.com>

fdkong committed:
Clean up executioner blocks for eigen problems

Close #17493

smharper committed:
Fix comment typo

Co-authored-by: Guillaume Giudicelli <guillaume.giudicelli@gmail.com>

roystgnr committed:
Style changes

jessecarterMOOSE committed:
Use more AutoLink shortcuts and flesh out descriptions, ref#14988

roystgnr committed:
Refactor to add InputParameters-accepting overload

This overload is a little trickier to use but is more in line with some
other MOOSE APIs.

loganharbour committed:
Add tests for exodus restart of array variables

refs #17121

jessecarterMOOSE committed:
Add some documentation for phase field anisotropy objects, ref#14988

loganharbour committed:
Merge pull request #17353 from lingzou/array-var-restart

Address ArrayVariable exodus restart issue

roystgnr committed:
Set _moose_app inside addMeshSubgenerator

This lets users be a little less picky about how they construct their
InputParameters without us hitting a failure at generate() time.

roystgnr committed:
Test both variadic and plain addMeshSubgenerator()

This way, if we ever add any more subtle parameters like _moose_app that
ought to be set from within addMeshSubgenerator() itself, we'll catch
the error in this test rather than in any user codes making use of the
InputParameters-accepting overload.

roystgnr committed:
git clang-format

roystgnr committed:
Update test/src/meshgenerators/StitchedSubgenerators.C

This is a test object only

Co-authored-by: Logan Harbour <loganhharbour@gmail.com>

lindsayad committed:
Add TwoMaterialPropertyInterface params to FVInterfaceKernel

rwcarlsen committed:
fix checkFileWriteable

rwcarlsen committed:
mod error message

roystgnr committed:
Better test directory name, requirement

Or a requirement that @loganharbour and I were both happy enough with,
at least.

roystgnr committed:
Rename StitchedSubgenerators -> TestSubgenerators

Logan is insistant that test objects should be clearly labeled as such.

Which makes sense, so I nominate him to fix the *other* 291 classes in
test/src/*/ that don't have "Test" in the name.

aeslaughter committed:
Restore complete SDD

(refs #15968)

aeslaughter committed:
Add test to get a reporter before it is declared

(refs #17468)

GiudGiud committed:
Merge pull request #17076 from smharper/scalar_mixing2

NSFV mixing length model for turbulent diffusion of scalars

YaqiWang committed:
this meshChanged() call seems unnecessary and does not break anything #17448

YaqiWang committed:
two new objects for array variables #17448

YaqiWang committed:
enhance this array kernel in test to support coupling with array variables #17448

YaqiWang committed:
extending array dg kernel and integrated bc to support HFEM #17448

YaqiWang committed:
add HFEM dg kernels and bcs and the tests #17448

aeslaughter committed:
Merge pull request #17467 from aeslaughter/sdd-15968

Restore complete SDD

rwcarlsen committed:
address review comments

rwcarlsen committed:
address review comments

rwcarlsen committed:
handle case where we can't find the installed tests

roystgnr committed:
Merge pull request #17458 from roystgnr/mesh_subgenerator_api

MeshGenerator::addMeshSubgenerator() api

lindsayad committed:
Fix potentially uninitialized warning in AutomaticMortarGeneration

zachmprince committed:
Making ReporterData check later #17468

bwspenc committed:
Minor fix to heading in SQA records page closes #17190

Commit:
Merge pull request #17474 from bwspenc/sqa_doc

Minor fix to heading in SQA records page

zachmprince committed:
Cloning reporters earlier in SamplerReporterTransfer #17468

YaqiWang committed:
document HFEM #17448

lindsayad committed:
Use quiet NaN instead of numeric_limits::max

Use of `std::numeric_limits<Real>::max()` opened us up to the
possibilities of generating infinities any time we multiplied that
number by something greater than 1. This can easily lead to invalid
floating point operations which could result in unknown behavior when we
don't trap fpes. When we did trap FPEs we would get obviously error out
(this happened for some Hertzian contact problems that Antonio is
investigating with heavily curved surfaces in which plenty of secondary
nodes fail to project onto the primary surface, meaning we end up in the
code branch that formerly generated potential infs). Instead of opening
up the possibility of inf, I now insert a quiet NaN (such that we don't
create floating point exceptions for intermediate operations) and then
check for `std::isnan` when deciding whether to drive the LM or the gap
to zero in `ApplyPenetrationConstraintLMMechanicalContact`. The formerly
failing cases that Antonio gave me are now working with this patch

zachmprince committed:
Creating surrogate trainer base class that loops through samples itself #16166

lindsayad committed:
Introduce member variables for tolerances in AutomaticMortarGeneration

In the future a developer can add APIs for changing these if they wish.
However, these tolerances have worked well for our mortar applications
thus far

lindsayad committed:
Address Fande review

Commit:
Merge pull request #15794 from arovinelli/break_mesh_hanging_node

Fix orphan nodes on 3D meshes for  BreakMeshByBlockGenerator

WilkAndy committed:
@loganharbour changes.  Fixes #17155

YaqiWang committed:
error with both mesh refinement and HFEM #17448

loganharbour committed:
Merge pull request #17343 from WilkAndy/geochem_units_17155

GeochemistryUnitConverter utility added, along with unit tests

loganharbour committed:
Merge pull request #16805 from jessecarterMOOSE/move_old_wiki_phase_field_docs

Start moving some phase field documentation from old wiki to new site

rwcarlsen committed:
Merge pull request #17428 from rwcarlsen/cp-tests

add --copy-tests flag for copying installed tests to curr dir.

permcody committed:
Fixing up generated heaader dependencies for submodules

closes #16637

YaqiWang committed:
add lower-d subdomain for all sides in MooseMesh #17448

permcody committed:
Merge pull request #17480 from permcody/header_fixup

Fixing up generated header dependencies for submodules

jiangwen84 committed:
fix parallel issue with SamplerTransientMultiApp and MultiAppCommandLineControl
ref #17486

lindsayad committed:
Assert that we have nodes on our mortar mesh

permcody committed:
Adding Git-LFS error test

refs #17407

permcody committed:
Adding utility to check for Git-LFS pointer files

refs #17407

lindsayad committed:
Only link to gperftools in profiling and opt modes

Closes #17490

lindsayad committed:
Follow Srinath's suggestion and encapsulate gap calcs

If we don't project onto a primary face then there should be no heat
transfer, e.g. our residual should just be lambda, e.g. we drive lambda
(representing the heat flux) to zero in regions of the interface where
we don't have an accompanying primary face

loganharbour committed:
Add error checking to UserObjectInterface and make more things const

Co-authored-by: April Novak <novak@berkeley.edu>

refs #17446 and #17512

fdkong committed:
Merge pull request #17189 from lindsayad/weighted-gap-16961

Weighted gap mortar mechanical contact implementation

vanwdani committed:
Create additional orders and families for material outputs
Refs #14823

lindsayad committed:
Limit mixed order weighted mortar gap test to sparse AD

Fixes up failure on next from merge of #17189

lindsayad committed:
Fix normal used for determining mortar element orientation

lindsayad committed:
Only use displacement derivatives on displaced mesh (#13488)

loganharbour committed:
Merge pull request #17498 from lindsayad/fixup-17189

Limit mixed order weighted mortar gap test to sparse AD

loganharbour committed:
Merge pull request #17494 from fdkong/clean_up_eigen_inputs

Clean up executioner blocks for eigen problems

aeslaughter committed:
Merge pull request #17487 from jiangwen84/fix_cmd_sampler_transient_multiapp

Fix a parallel issue with SamplerTransientMultiApp and MultiAppCommandL…

YaqiWang committed:
issue an error for now with distributed mesh #17448

lindsayad committed:
Apply snesmf_reuse_base = false as indicated by devel failures

aeslaughter committed:
Add date/time capability to MooseDocs

(refs #17505)

lindsayad committed:
Update default 'c' value for ApplyPenetrationLMMechanicalContact

Commit:
Merge pull request #17491 from lindsayad/gperf-link-17490

Only link to gperftools in profiling and opt modes

loganharbour committed:
Move threaded copy designator to UserObject

refs #17512

aeslaughter committed:
Add gitutils extension

(closes #17505)

loganharbour committed:
Merge pull request #17380 from loganharbour/scalar_kernel_u_old

Remove _u_old from ScalarKernel/AuxScalarKernel and const-ify scalar vars

zachmprince committed:
Separating getting reporter values and sampler data #16166

YaqiWang committed:
move a function into source file #17448

vanwdani committed:
Review changes

loganharbour committed:
Merge pull request #17496 from aprilnovak/fluid-api

Improve UserObjectInterface errors

aeslaughter committed:
Update working of gitutils extension docs

(refs #17505)

WilkAndy committed:
allow specification of bulk composition with or without kinetic contributions

Refs #17144

Commit:
Merge pull request #17447 from YaqiWang/hfem

Hybrid FEM support

WilkAndy committed:
changes to doco and tests from the with_kinetic changes

Fixes #17144

YaqiWang committed:
fix an bug for assembling jacobian on boundary with HFEM #17448

aeslaughter committed:
Merge pull request #17513 from aeslaughter/date-17505

Add date and git commit to MooseDocs

lindsayad committed:
Merge pull request #17518 from YaqiWang/hfem

Fix a bug in PR 17447

lindsayad committed:
Fix interior_parent bug in Assembly::computeFaceMap

lindsayad committed:
Reduce nl_abs_div_tol to catch better solves

lindsayad committed:
Register ray-tracing segment mesh as distributed from start

lindsayad committed:
Apply suggestions from code review, round 2

Co-authored-by: Guillaume Giudicelli <guillaume.giudicelli@gmail.com>

lindsayad committed:
Use upstream capability to restart scalar variables

David added a first class capability to copy scalar solutions in
libmesh/libmesh#2123 so we should leverage that here instead of trying
to reinvent the wheel (apparently unsuccessfully as indicated by #17471)

Closes #17471

lindsayad committed:
Issue #17471 test

lindsayad committed:
Merge pull request #17522 from lindsayad/scalar-restart-17471

Use upstream capability to restart scalar variables

lindsayad committed:
Merge pull request #17450 from lindsayad/libmesh-update

Update libmesh

lindsayad committed:
Apply suggestions from code review, round 3

- Add test of first order convergence without an interface kernel

Co-authored-by: Guillaume Giudicelli <guillaume.giudicelli@gmail.com>

lindsayad committed:
Fixup for #17522

lindsayad committed:
Apply prereq to prevent interference during recover

GiudGiud committed:
Merge pull request #17399 from lindsayad/as-restart-multiapps-17078

Add scaling vector if we are doing automatic scaling in MooseVariableBase::initialSetup

GiudGiud committed:
Merge pull request #17535 from lindsayad/fix-17522

Fixup for #17522

dschwen committed:
Merge pull request #17514 from vanwdani/TMA_Material_output

Create additional orders and families for material outputs

GiudGiud committed:
Merge pull request #17449 from lindsayad/fv-ik-17087

Add FVInterfaceKernels

Commit:
fix samplertransientmultiapp with multiplecommandlinecontrol when #mpi is greater than #samples
ref #17548

lindsayad committed:
Restrict HFEM tests to when we have mumps

We got a failure on libmesh testing when mumps is not available
[here](https://civet.inl.gov/job/720896/).

Refs #17447

loganharbour committed:
Verify node and point neighbor routines using libMesh routines

refs #17552

loganharbour committed:
Merge pull request #17516 from WilkAndy/ini_kin_17144

Geochemistry: ability to specify bulk constraints with or without kinetic contributions

permcody committed:
Addressing comments

refs #17407

lindsayad committed:
Merge pull request #17252 from lindsayad/th-contact-jac-13488

Approximate gap Jacobians in GapConductanceConstraint

Commit:
Merge pull request #17425 from aprilnovak/less-restrictive-error

Allow multiple element sides on same sideset for elemental boundary auxkernel

Commit:
Merge pull request #17554 from lindsayad/restrict-mumps-17447

Restrict HFEM tests to when we have mumps

loganharbour committed:
Merge pull request #17553 from loganharbour/verify_rt_neighbors

Verify ray_tracing node and point neighbor routines

andrsd committed:
Merge pull request #17489 from permcody/git_lfg

Git LFS File Test

aeslaughter committed:
Merge pull request #17549 from jiangwen84/fix_cmd_sampler_transient_multiapp

Fix samplertransientmultiapp issue when number of samples is less than MPI process

loganharbour committed:
Make RayTracingUtils::mixedPackSize() c++11 compatibile

refs #17578

loganharbour committed:
Separate declarations and implementations in RayTacingPackingUtils

refs #17578

roystgnr committed:
Require MUMPS in all HFEM tests that use it

Fixes #17622

A better solution would be to not use MUMPS here, IMHO.  Surely HFEM
matrix conditioning isn't so wonky that we can't handle a small test
case without a direct solver?

loganharbour committed:
Merge pull request #17582 from loganharbour/fix_rt_constexpr

Clean RayTracingPackingUtils and make c++11 compatible

aeslaughter committed:
Merge pull request #17485 from zachmprince/surrogate_template

Creating intermediate class for STM trainers

roystgnr committed:
Merge pull request #17623 from roystgnr/hfem_robin_displaced_mumps

Require MUMPS in all HFEM tests that use it

Commit:
Trigger a build

Testing dynamic module loading

Commit:
Merge branch '000-test-meowbuild' of github.com:milljm/moose into 000-test-meowbuild

milljm commented:
is what happens when do exactly what `git` tells you to do....

Commit:
Triggering a build

Trigger a new build to test WASP target.

Refs #000

milljm commented:
hah, I guess Civet no longer wants to work on this PR. Closing. Will create another.
